### ISSUE TITLE
Impossible to translate sorting labels from Facetedsearch module

### ISSUE THREAD

**Issue by Prestaworks** [PrestaShop/PrestaShop#35237]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In Facetedsearch module, there are labels for selecting sort order in product list.
Sales, highest to lowest, Relevance, Name, A to Z and so on.
Some of these are missing translations, like Reference, A to Z for instance and it is not listed in the BO.
The reason for this is that the labels loaded from the module are connected to the translation domain Shop.Theme.Catalog

### Expected behavior

It should be possible to select faceted search module in translations and change these labels.

### Steps to reproduce

1. Install facetedsearch module.
2. Install Swedish language.
3. Go to category in frontend, "Reference, A to Z" is shown in dropdown for sortorder
4. Try to change this.

### PrestaShop version(s) where the bug happened

8.1.3

### PHP version(s) where the bug happened

8.1.12

### If your bug is related to a module, specify its name and its version

facetedsearch 3.14.1

### Your company or customer's name goes here (if applicable).

Prestaworks AB

**Comment by florine2623** [comment]
Hello @Prestaworks ,

I reproduce the issue with PrestaShop version `8.1.x` and Faceted search module `v3.12.1` and `v3.14.1`, I'll add this to the backlog so it can be fixed.

https://github.com/PrestaShop/PrestaShop/assets/16019289/ee0d1cd6-d54c-41db-bbec-42abdf42603d

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by Hlavtox** [comment]
I can translate the labels just fine. The reference sort can't be translated because it doesn't discover it and it will be part of the language pack when 8.1.4 is released. This is a known limitation of the translation system.

![Snímek obrazovky 2024-02-02 121605](https://github.com/PrestaShop/PrestaShop/assets/6097524/079e6e01-7ad8-42ef-a8a6-793d8b1a4bd3)
![Snímek obrazovky 2024-02-02 121613](https://github.com/PrestaShop/PrestaShop/assets/6097524/4e33bb35-3fae-4edd-9872-6c354748f473)

**Comment by Prestaworks** [comment]
ok, so the module can't be 100% translated for earlier versions than an unreleased version and that is fine? If so, go ahead and close the report and close the PR.

**Comment by Hlavtox** [comment]
@Prestaworks Yeah, sucks, but that's how it is. It's the same situation with other strings and other modules. The same problem was with "Available" value in filters.

- First problem is that the translation system doesn't discover all new translations in some cases.
- Second problem is that XLF files for modules are distributed with the core, not the modules. So when you change something or add new strings, only the latest core version get the translation.

You can work around it quite easily, if you know:
- Insert the translation manually into `ps_translations` table.
- Delete `var/cache/prod/translations` folder.
- Voila, your translation is working.

**Comment by Prestaworks** [comment]
Yeah I know, I can solve it, you too, but the average merchant can't :(

Maybe the best thing would be to redo the translations so there are no dividing between modules/front/BO and just all domains from all files are displayed in a big tree list. Would probobly be easier to find what you are looking for that way.

**Comment by Hlavtox** [comment]
@Prestaworks Well, I try to convince somebody from the maintainer team for years to solve this, that it's an important flaw, but haven't been successful. 🤷‍♂️

**Comment by jolelievre** [comment]
I don't know the translation system well enough, but I'd guess the limitation is because the UX favors using the XLF default catalog because scanning aaaallll the code would be too long every time you need to translate something.

That said I agree with you that it's a pain that we can't edit wordings as long as the default catalog is not updated, maybe we could handle a systematic/automatic scan of all the files but then it should be cached. So the first init would take a while (potentially 1 or 2 minutes I'm not sure) but then you have it all. And when some new wordings are added you just need to clear the cache, but since new wordings are likely to come with update modules or update core these operations are already followed by a cache clear so it could work.

**Comment by Prestaworks** [comment]
Sounds like a good way to handle it @jolelievre

**Comment by MatShir** [comment]
@Hlavtox, it's something I want to improve. However, as you know @Hlavtox, there are many ideas and issues for a few project members. I'm considering whether it's worth investing in a dedicated search instead of loading all the data for specific translation. 
I've identified two use cases from the merchants' translation needs:
- For a new project or module, I aim to translate the entire module or FO. In this scenario, the solution would be to display all or export/import all. Loading a large amount of data should not be an issue, but I would recommend importing it.
- For specific translation: I want to make changes to wording (because I forgot, or the current translation isn't suitable). In this case, it would only require searching across all domains directly without the need to navigate through domains, etc and load all the data and so on.

**Comment by Hlavtox** [comment]
@MatShir I don't understand what you are saying. The solution is this:
- Core must discover translations in modules. It's a matter of ms, max single seconds. This will solve the issue with not being able to translate the strings that are not in the current XLF catalog.
- We must move XLFs to module repositories so they are distributed with the module not with the core. Core versions 1.7.8+ will work properly with them and load them. This will solve the issue of not having up-to-date XLF catalog after updating a module, because we update and distribute them with the core.

**Comment by Hlavtox** [comment]
I made a little test on `ps_newproducts` module.

### Translations in PHP file
$a = $this->trans('TESTPHP', [], 'Modules.Newproducts.Shop'); ✅ Translatable using module translations
$a = $this->trans('TESTPHP', [], 'Shop.Theme.Catalog'); 🔴  Not available, the discovery probably found this, but doesn't offer for translation.

### Translation in `/modules/ps_newproducts/views/templates/hook/ps_newproducts.tpl`
{l s='TESTTPLMODULES' d='Modules.Newproducts.Shop'} ✅ Translatable using module translations
{l s='TESTTPLMODULES' d='Shop.Theme.Catalog'} 🔴 Not available, the discovery probably found this, but doesn't offer for translation.

### Translation in `/themes/classic/modules/ps_newproducts/views/templates/hook/ps_newproducts.tpl`
{l s='TESTTPLTHEMEMODULES' d='Modules.Newproducts.Shop'} ✅ Translatable using theme translations
{l s='TESTTPLTHEMEMODULES' d='Shop.Theme.Catalog'} ✅ Translatable using theme translations
